### PR TITLE
CAMEL-12651 - Allow to override headers serializing and deserializing

### PR DESCRIPTION
**Pull request by tdanylchuk**

### CODE REVIEW

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java [hunk 1]
```diff
@@ -16,21 +16,14 @@
...
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Properties;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
```
Comment (oscerd) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java]: Please avoid importing stuff  with *

File: components/camel-kafka/src/test/java/org/apache/camel/component/kafka/serde/DefaultKafkaHeaderSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public DefaultKafkaHeaderSerializerTest(Object value, byte[] expectedResult) {
+        this.value = value;
+        this.expectedResult = expectedResult;
+    }
+
+    @Test
+    public void serialize() {
+        byte[] result = serializer.serialize("someKey", value);
+
+        System.out.println(Arrays.toString(result));
```
Comment (oscerd) [inline components/camel-kafka/src/test/java/org/apache/camel/component/kafka/serde/DefaultKafkaHeaderSerializerTest.java]: Use a log and don't print on standard out please

Comment (tdanylchuk) [reply components/camel-kafka/src/test/java/org/apache/camel/component/kafka/serde/DefaultKafkaHeaderSerializerTest.java]: oh, forgot to remove

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java (hunk 1 above)
Comment (tdanylchuk) [reply components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java]: done

**Comment by davsclaus** [comment]
Good work. There is possible a checkstyle error, which you can see via

   mvn compile -P sourcecheck

And also if possible the spring-boot starter JAR has been updated in the mean time, so you may want to rebase this PR on master and rebuild the camel-kafka component so it generates updated spring-boot JAR source.

**Comment by tdanylchuk** [comment]
done, 
I was also curious, if there any possibility to pick bean from spring context in such available instead of directly setting it to rout URI? (question from gitter)

**Comment by oscerd** [comment]
Merged on master. Thanks.

**Pull request closed without merging**